src/sentry/static/sentry/app/views/ruleEditor/index.jsx

```diff
@@ -136,13 +128,35 @@ const RuleEditor = createReactClass({
     return !!error[field];
   },
 
+  updateRule(prop, val) {
+    let rule = {...this.state.rule};
+    rule[prop] = val;
+    this.setState({
```

Comment:
I would use setState updater callback here since we're updating state based on a previous state:

`this.setState(state => {
    let rule = {...state.rule};
    rule[prop] = val;
    return {rule};
});`